File: app/src/main/java/net/mieczkowski/yelpbusinessexample/recyclerAdapters/searchBusiness/SearchBusinessAdapter.java
package net.mieczkowski.yelpbusinessexample.recyclerAdapters.searchBusiness;

import android.view.ViewGroup;

import net.mieczkowski.yelpbusinessexample.R;
import net.mieczkowski.yelpbusinessexample.models.business.YelpBusiness;
import net.mieczkowski.yelpbusinessexample.recyclerAdapters.base.BaseAdapter;
import net.mieczkowski.yelpbusinessexample.recyclerAdapters.base.BaseViewHolder;

import java.util.List;

/**
 * Created by Josh Mieczkowski on 10/22/2017.
 */

public class SearchBusinessAdapter extends BaseAdapter<YelpBusiness> {

    private static final int EMPTY_VIEW = 0;
    private static final int BUSINESS_VIEW = 1;

    public SearchBusinessAdapter(List<YelpBusiness> objects) {
        super(objects);
    }

    @Override
    public int getItemViewType(int position) {
        if(objects.isEmpty()){
            return EMPTY_VIEW;
        }else{
            return BUSINESS_VIEW;
        }
    }

    @Override
    public BaseViewHolder<YelpBusiness> onCreateViewHolder(ViewGroup parent, int viewType) {
        switch (viewType){
            case EMPTY_VIEW:
                return new SearchBusinessEmptyViewHolder(getView(parent, R.layout.row_search_no_results));
            default:
                return new SearchBusinessViewHolder(getView(parent, R.layout.row_seach_business));
        }
    }

    @Override
    public int getItemCount() {
        return objects.isEmpty() ? 1 : objects.size();
    }
}

File: app/src/main/java/net/mieczkowski/yelpbusinessexample/recyclerAdapters/searchBusiness/SearchBusinessAdapter.kt
package net.mieczkowski.yelpbusinessexample.recyclerAdapters.searchBusiness

import android.view.ViewGroup
import com.bluelinelabs.conductor.Router
import net.mieczkowski.dal.services.businessLookupService.models.YelpBusiness

import net.mieczkowski.yelpbusinessexample.R
import net.mieczkowski.yelpbusinessexample.controllers.DetailController
import net.mieczkowski.yelpbusinessexample.exts.show
import net.mieczkowski.yelpbusinessexample.recyclerAdapters.base.BaseAdapter
import net.mieczkowski.yelpbusinessexample.recyclerAdapters.base.BaseViewHolder

/**
 * Created by Josh Mieczkowski on 10/22/2017.
 */

class SearchBusinessAdapter(objects: List<YelpBusiness>, private val router: Router) : BaseAdapter<YelpBusiness>(objects) {

    private val EMPTY_VIEW = 0
    private val BUSINESS_VIEW = 1

    override fun getItemViewType(position: Int): Int {
        return if (objects.isEmpty()) {
            EMPTY_VIEW
        } else {
            BUSINESS_VIEW
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder<YelpBusiness> {
        return when (viewType) {
            EMPTY_VIEW -> SearchBusinessEmptyViewHolder(getView(parent, R.layout.row_search_no_results))
            else -> SearchBusinessViewHolder(getView(parent, R.layout.row_seach_business))
        }
    }

    override fun getItemCount(): Int {
        return if (objects.isEmpty()) 1 else objects.size
    }

    override fun onRowClick(position: Int) {
        super.onRowClick(position)

        DetailController(objects[position]).show(router)
    }
}
